orc/src/main/java/org/apache/iceberg/orc/OrcMetrics.java

```diff
@@ -0,0 +1,154 @@
+          }
+        }
+      }
+
+      return new Metrics(orcReader.getNumberOfRows(),
+          columSizes,
+          valueCounts,
+          Maps.newHashMap(),
+          lowerBounds,
+          upperBounds);
+    } catch (IOException ioe) {
+      throw new RuntimeIOException(ioe, "Failed to read footer of file: %s", file);
+    }
+  }
+
+  private static Optional<ByteBuffer> fromOrcMin(Types.NestedField column,
+                                                    OrcProto.ColumnStatistics columnStats) {
+    ByteBuffer min = null;
+    if (columnStats.hasIntStatistics()) {
+      if (column.type().typeId() == Type.TypeID.INTEGER) {
+        min = toByteBuffer(column.type(), (int) columnStats.getIntStatistics().getMinimum());
+      } else {
+        min = toByteBuffer(column.type(), columnStats.getIntStatistics().getMinimum());
+      }
+    } else if (columnStats.hasDoubleStatistics()) {
+      min = toByteBuffer(column.type(), columnStats.getDoubleStatistics().getMinimum());
+    } else if (columnStats.hasStringStatistics()) {
+      min = toByteBuffer(column.type(), columnStats.getStringStatistics().getMinimum());
+    } else if (columnStats.hasDecimalStatistics()) {
+      min = toByteBuffer(column.type(), columnStats.getDecimalStatistics().getMinimum());
+    } else if (columnStats.hasDateStatistics()) {
+      min = toByteBuffer(column.type(), columnStats.getDateStatistics().getMinimum());
+    } else if (columnStats.hasTimestampStatistics()) {
+      min = toByteBuffer(column.type(), columnStats.getTimestampStatistics().getMinimum());
+    }
+    return Optional.ofNullable(min);
+  }
+
+  private static Optional<ByteBuffer> fromOrcMax(Types.NestedField column,
+                                                    OrcProto.ColumnStatistics columnStats) {
+    ByteBuffer max = null;
+    if (columnStats.hasIntStatistics()) {
+      if (column.type().typeId() == Type.TypeID.INTEGER) {
+        max = toByteBuffer(column.type(), (int) columnStats.getIntStatistics().getMaximum());
+      } else {
+        max = toByteBuffer(column.type(), columnStats.getIntStatistics().getMaximum());
+      }
+    } else if (columnStats.hasDoubleStatistics()) {
+      max = toByteBuffer(column.type(), columnStats.getDoubleStatistics().getMaximum());
+    } else if (columnStats.hasStringStatistics()) {
+      max = toByteBuffer(column.type(), columnStats.getStringStatistics().getMaximum());
+    } else if (columnStats.hasDecimalStatistics()) {
+      max = toByteBuffer(column.type(), columnStats.getDecimalStatistics().getMaximum());
+    } else if (columnStats.hasDateStatistics()) {
+      max = toByteBuffer(column.type(), columnStats.getDateStatistics().getMaximum());
+    } else if (columnStats.hasTimestampStatistics()) {
+      max = toByteBuffer(column.type(), columnStats.getTimestampStatistics().getMaximum());
+    }
+    return Optional.ofNullable(max);
+  }
+
+  static Map<Integer, ?> fromBufferMap(Schema schema, Map<Integer, ByteBuffer> map) {
```

Comment:
Seems this is only used in the test. Should this then be in the test class? Or do u see this being used in other places

Reply:
I'll make it for testing only since I don't see any other use yet.